_posts/2017-05-26-deploying-create-react-app-to-s3.markdown

```diff
@@ -0,0 +1,267 @@
+---
+layout: post
+title:  "Deploying create-react-app via S3"
+date:   2017-06-01 00:00:00 -0000
+categories: react devops
+author: Zach Cotter
+---
+
+At Forward Financing, we've started using
+[create-react-app][create-react-app]
+as a starter kit for all of our new React projects. Create-React-App is awesome
+because it comes with a one size fits all webpack configuration via the
+`react-scripts` package. All you have to do is `npm run build` and you have a
+ready to ship `/build` folder with minified production asset files.
+
+The first project we did this with was a front-end for an application used internally to manage our user's roles and access to pages across our suite
+of apps. The backend server was our standalone single sign on service,
+written in Elixir+Phoenix, which had some existing server rendered views. An
+API for the new React frontend was built into that backend service.
+
+We had a few requirements for the deployment process of the React app:
+
+- The compiled javascript+css assets needed to be hosted somewhere cheaply and
+efficiently.
+- The react app need to live at a subresource of our existing single sign on
+application, specifically `login.forwardfinancing.com/panel/user_roles`.
+- The react app needed to have a separate deploy process from the backend app
+
+### Hosting the React build folder
+
+Lots of people seem to be using Express to host their react apps on a shared
+virtual server provider like Heroku or Digital Ocean. This doesn't make a lot of
+sense to me because you are paying ~$5/month just to serve up a few static
+files from a dedicated cloud server. I also am not a fan of using javascript as
```

Comment:
If you can keep it brief, add a sentence here on why you don't like JS as a web server.